builder/dockerfile/copy.go

```diff
@@ -315,6 +315,9 @@ func (o *copier) copyWithWildcards(origPath string) ([]copyInfo, error) {
 func copyInfoForFile(source builder.Source, path string) (copyInfo, error) {
 	fi, err := remotecontext.StatAt(source, path)
 	if err != nil {
+		if errors.Is(err, os.ErrNotExist) {
+			return copyInfo{}, errors.Wrapf(os.ErrNotExist, "%s not found or excluded by .dockerignore", path)
```

Comment:
I guess one way would be to use `PathError` in this case. eg. like in https://github.com/tonistiigi/fsutil/blob/master/walker.go#L28

Reply:
Ah, yes, that's probably the correct way. I'll have a try what that looks like; does that approach allow a custom error message and still being detected as an `os.IsNotExist()` ? 🤔

Reply:
It uses this formatting https://golang.org/src/os/error.go?s=1379:1413#L36 , you can customize the prefix. It still works in `errors.Is(err, os.ErrNotExist)` and you could also get the typed path value from the error.

Reply:
Hm, so, I looked, and not entirely sure how you're suggesting to add the prefix; do you mean setting the custom message as `Op`? So 

```suggestion
			return copyInfo{}, errors.WithStack(&os.PathError{Op: "file not found or excluded by .dockerignore:", Path: path, Err: os.ErrNotExist})
```

and/or passing the original error as `Err`;

```suggestion
			return copyInfo{}, errors.WithStack(&os.PathError{Op: "file not found or excluded by .dockerignore:", Path: path, Err: err})
```

(doing so would include both the "new" and the "old/confusing" error in the output, so that didn't really work)


Or, set `Op: "stat"`, and used `errors.Wrap` to pass the custom description;

```suggestion
			return copyInfo{}, errors.Wrap(&os.PathError{Op: "stat", Path: path, Err: os.ErrNotExist}, "file not found or excluded by .dockerignore")
```